### PR TITLE
tools/tidb-binlog: update binlog deploy info

### PR DESCRIPTION
**Pull request by anotherrachel**
### What is changed, added or deleted? 

- Delete unused configure toml files.
- Update toml and config indexes for Drainer & Pump.
- Add more notes to help customers to avoid or handle errors during deployment and maintenance.     These notes are mainly about the consistence of the sql_mode and compatibility of SQL between upstream and downstream databases.
- Remove extra "$" in the code block.
- Make code block copyable.

### What is the related PR or file link(s)? 
https://github.com/pingcap/docs-cn/pull/1879
https://github.com/pingcap/docs/pull/1585/

### Which version does your change affect? 
`deploy.md`: dev, v2.1, v3.0
`faq.md`: v2.1, v3.0

### CODE REVIEW

**Comment by anotherrachel** [comment]
@lichunzhu @TomShawn PTAL, thanks!

File: v3.0/reference/tools/tidb-binlog/deploy.md [hunk 1]
```diff
@@ -567,3 +615,5 @@ The following part shows how to use Pump and Drainer based on the nodes above.
 > - Make sure that the TiDB Binlog service is enabled in all TiDB instances in a same cluster, otherwise upstream and downstream data inconsistency might occur during data replication. If you want to temporarily run a TiDB instance where the TiDB Binlog service is not enabled, set `run_ddl=false` in the TiDB configuration file.
 > - Drainer does not support the `rename` DDL operation on the table of `ignore schemas` (the schemas in the filter list).
 > - If you want to start Drainer in an existing TiDB cluster, generally you need to make a full backup of the cluster data, obtain `savepoint`, import the data to the target database, and then start Drainer to replicate the incremental data from `savepoint`.
+> - When the downstream database is TiDB or MySQL, the  `sql_mode` in both upstream and downstream database should be consistent. In other words, the `sql_mode` should be the same when each SQL statement is executed in the upstream and synchronized in the downstream. You can execute `select @@sql_mode;` statement in upstream and downstream respectively to compare `sql_mode`.
```
Comment (lichunzhu) [inline v3.0/reference/tools/tidb-binlog/deploy.md]:
```suggestion
> - When the downstream database is TiDB or MySQL, the `sql_mode` in both upstream and downstream database should be consistent. In other words, the `sql_mode` should be the same when each SQL statement is executed in the upstream and synchronized in the downstream. You can execute `select @@sql_mode;` statement in upstream and downstream respectively to compare `sql_mode`.
```

File: v3.0/reference/tools/tidb-binlog/deploy.md [hunk 2]
```diff
@@ -391,15 +422,16 @@ The following part shows how to use Pump and Drainer based on the nodes above.
         -c int
             the number of the concurrency of the downstream for replication. The bigger the value, the better throughput performance of the concurrency (1 by default).
         -cache-binlog-count int
-            the limit on the number of binlog items in the cache (65536 by default)
+            the limit on the number of binlog items in the cache (512 by default)
+            If a single binlog upstream causes OOM in Drainer, try reduce the number of binlog items to reduce memory usage.
```
Comment (lichunzhu) [inline v3.0/reference/tools/tidb-binlog/deploy.md]: ditto
```suggestion
            If a large binlog upstream causes OOM in Drainer, try reduce the number of binlog items to reduce memory usage.
```

File: dev/reference/tools/tidb-binlog/deploy.md [hunk 3]
```diff
@@ -576,3 +614,5 @@ The following part shows how to use Pump and Drainer based on the nodes above.
 > - Make sure that the TiDB Binlog service is enabled in all TiDB instances in a same cluster, otherwise upstream and downstream data inconsistency might occur during data replication. If you want to temporarily run a TiDB instance where the TiDB Binlog service is not enabled, set `run_ddl=false` in the TiDB configuration file.
 > - Drainer does not support the `rename` DDL operation on the table of `ignore schemas` (the schemas in the filter list).
 > - If you want to start Drainer in an existing TiDB cluster, generally you need to make a full backup of the cluster data, obtain `savepoint`, import the data to the target database, and then start Drainer to replicate the incremental data from `savepoint`.
+> - When the downstream database is TiDB or MySQL, the  `sql_mode` in both upstream and downstream database should be consistent. In other words, the `sql_mode` should be the same when each SQL statement is executed in the upstream and synchronized in the downstream. You can execute `select @@sql_mode;` statement in upstream and downstream respectively to compare `sql_mode`.
```
Comment (TomShawn) [inline dev/reference/tools/tidb-binlog/deploy.md]:
```suggestion
> - When the downstream database is TiDB or MySQL, ensure that the `sql_mode` in the upstream and downstream databases are consistent. In other words, the `sql_mode` should be the same when each SQL statement is executed in the upstream and replicated to the downstream. You can execute the `select @@sql_mode;` statement in the upstream and downstream respectively to compare `sql_mode`.
```

File: dev/reference/tools/tidb-binlog/deploy.md [hunk 4]
```diff
@@ -576,3 +614,5 @@ The following part shows how to use Pump and Drainer based on the nodes above.
...
+> - Drainer also fails to synchronize when there are DDL statements that upstream database supports but downstream database does not. An example is synchronizing `CREATE TABLE t1(a INT) ROW_FORMAT=FIXED;` statement when the downstream database MySQL uses the InnoDB engine. In this case, you can configure [skipping transactions](/dev/reference/tools/tidb-binlog/faq.md#what-can-i-do-when-some-ddl-statements-supported-by-the-upstream-database-cause-error-when-executed-in-the-downstream-database) in Drainer, and manually execute compatible statements in the downstream database.
```
Comment (TomShawn) [inline dev/reference/tools/tidb-binlog/deploy.md]:
```suggestion
> - When a DDL statement is supported in the upstream but incompatible with the downstream, Drainer fails to replicate data. An example is to replicate the `CREATE TABLE t1(a INT) ROW_FORMAT=FIXED;` statement when the downstream MySQL uses the InnoDB engine. In this case, you can configure [skipping transactions](/dev/reference/tools/tidb-binlog/faq.md#what-can-i-do-when-some-ddl-statements-supported-by-the-upstream-database-cause-error-when-executed-in-the-downstream-database) in Drainer, and manually execute the compatible statement in the downstream database.
```

File: v2.1/reference/tools/tidb-binlog/deploy.md [hunk 5]
```diff
@@ -33,28 +33,16 @@ In environments of development, testing and production, the requirements on serv
 
     | tidb-ansible branch | TiDB version | Note |
     | :------------------- | :------------ | :---- |
-    | release-2.0 | 2.0 version | The latest 2.0 stable version. You can use it in the production environment. |
     | release-2.1 | 2.1 version | The latest 2.1 stable version. You can use it in the production environment (recommended). |
```
Comment (TomShawn) [inline v2.1/reference/tools/tidb-binlog/deploy.md]:
```suggestion
    | release-2.1 | 2.1 version | The latest 2.1 stable version. You can use it in the production environment. |
```

File: dev/reference/tools/tidb-binlog/deploy.md [hunk 6]
```diff
@@ -399,15 +419,16 @@ The following part shows how to use Pump and Drainer based on the nodes above.
         -c int
             the number of the concurrency of the downstream for replication. The bigger the value, the better throughput performance of the concurrency (1 by default).
         -cache-binlog-count int
-            the limit on the number of binlog items in the cache (65536 by default)
+            the limit on the number of binlog items in the cache (512 by default)
+            If a single binlog in the upstream causes OOM in Drainer, try reduce the number of binlog items to reduce memory usage.
```
[the remark this answers is not in the document]
Comment (lichunzhu) [reply dev/reference/tools/tidb-binlog/deploy.md]: Maybe `item` is better than `file` here?

[the remark this answers is not in the document]
Comment (TomShawn) [reply dev/reference/tools/tidb-binlog/deploy.md]: Yes.

**Comment by anotherrachel** [comment]
All comments addressed. PTAL again, thanks! @TomShawn @lichunzhu

File: v3.0/reference/tools/tidb-binlog/deploy.md [hunk 7]
```diff
@@ -147,9 +156,14 @@ In environments of development, testing and production, the requirements on serv
...
+    {{< copyable "shell-regular" >}}
+
     ```bash
-    $ cd /home/tidb/tidb-ansible
-    $ resources/bin/binlogctl -pd-urls=http://127.0.0.1:2379 -cmd generate_meta
+    cd /home/tidb/tidb-ansible
+    ```
+
+    ```
+    resources/bin/binlogctl -pd-urls=http://127.0.0.1:2379 -cmd generate_meta
```
Comment (lichunzhu) [inline v3.0/reference/tools/tidb-binlog/deploy.md]: ditto

File: dev/reference/tools/tidb-binlog/deploy.md [hunk 8]
```diff
@@ -154,9 +156,14 @@ In environments of development, testing and production, the requirements on serv
...
+    {{< copyable "shell-regular" >}}
+
     ```bash
-    $ cd /home/tidb/tidb-ansible
-    $ resources/bin/binlogctl -pd-urls=http://127.0.0.1:2379 -cmd generate_meta
+    cd /home/tidb/tidb-ansible
+    ```
+
+    ```
+    resources/bin/binlogctl -pd-urls=http://127.0.0.1:2379 -cmd generate_meta
```
Comment (lichunzhu) [inline dev/reference/tools/tidb-binlog/deploy.md]: Should this line be moved to last block?

**Pull request merged**